### PR TITLE
Vacms 13222 display where field for events

### PR DESCRIPTION
**Pull request by chriskim2311**
## Summary
For the event listing pages we have accounted for new address format of the Where field. We have now added addresses for VA locations and added directions link to google maps for all events that include an address VA location or not.

## Related issue(s)
[#13222](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13222)

## Testing done
Went through event listings to verify that all event types have the correct address showing and that online events still show that they are specifically online. 

Verified that all directions link are also working for listing and verified that address to location is being passed properly. 

Testing URL 
/outreach-and-events/events/

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | Outreach___Events___Outreach_Events___Veterans_AffairsbeforeMobile |  Outreach___Events___Outreach_Events___Veterans_Affairsafter1 |
| Desktop | Outreach___Events___Outreach_Events___Veterans_Affairsbefore |  Outreach___Events___Outreach_Events___Veterans_Affairs |

## What areas of the site does it impact?

Outreach Events listing page

## Acceptance criteria
- [ ] Address is presented as described, including Directions link 
- [ ] Optional building/floor/room text is displayed if provided by CMS
- [ ] Google maps directions link is present / working for all addresses
- [ ] VA facilities additionally display the name of the VA facility
  - [ ] If the VA facility has a web page (eg VAMC), the facility name is hyperlinked to that page (see Eng Notes above)
  - [ ] Otherwise, there is no hyperlink (eg VC Outstation)
- [ ] Design review required
- [ ] A11y review required

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

File: src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx [hunk 1]
```diff
@@ -2,20 +2,36 @@ import React from 'react';
...
-const WhereContent = ({ fieldLocation, fieldType, derivedLocations }) => {
+const WhereContent = ({
+  event,
+  fieldLocation,
+  fieldType,
+  derivedLocations,
+}) => {
   if (fieldType === 'online') return 'This is an online event.';
+  const locationAddress =
+    derivedLocations?.length > 0 ? derivedLocations.join(' ') : '';
```
Comment (randimays) [inline src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx:13]: Could this be simplified as 
```
derivedLocations?.length ? derivedLocations.join(' ') : '';
```

File: src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx [hunk 2]
```diff
@@ -2,20 +2,36 @@ import React from 'react';
...
   return (
     <>
       <a href={fieldLocation?.entity?.entityUrl.path}>
         {fieldLocation?.entity?.title}
       </a>
+      <p className="vads-u-margin--0" key={location}>
+        {event?.fieldLocationHumanreadable}
```
Comment (randimays) [inline src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx:20]: Is this field often not defined? If so, could we make rendering the paragraph conditional upon the existence of this field?

File: src/applications/static-pages/events/helpers/index.js [hunk 3]
```diff
@@ -351,27 +351,45 @@ export const deriveEventLocations = event => {
     return locations;
   }
 
-  if (event?.fieldLocationHumanreadable) {
-    locations.push(event?.fieldLocationHumanreadable);
-  }
+  if (event?.fieldFacilityLocation?.entity?.fieldAddress) {
+    const fieldFacilityEntity = event?.fieldFacilityLocation?.entity;
+    if (fieldFacilityEntity?.fieldAddress?.addressLine1) {
```
Comment (randimays) [inline src/applications/static-pages/events/helpers/index.js:356]: A lot of references to `fieldFacilityEntity?.fieldAddress` here. Maybe we could destruct `addressLine1`, addressLine2`, etc. off that variable at the top to lower the cognitive load here?

File: src/applications/static-pages/events/helpers/index.unit.spec.js [hunk 4]
```diff
@@ -315,12 +315,6 @@ describe('deriveEventLocations', () => {
     expect(deriveEventLocations()).to.deep.equal([]);
   });
 
-  it('handles when fieldLocationHumanreadable is truthy', () => {
```
Comment (randimays) [inline src/applications/static-pages/events/helpers/index.unit.spec.js:318]: Curious - why was this test removed?

**Review by powellkerry — APPROVED** [review]
Once the comments from Randi are resolved this looks good to me.

File: src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx [hunk 5]
```diff
@@ -2,20 +2,36 @@ import React from 'react';
...
           {derivedLocations?.map(location => (
             <p className="vads-u-margin--0" key={location}>
               {location}
             </p>
           ))}
+          <a
+            href={`https://maps.google.com?saddr=Current+Location&daddr=${locationAddress}`}
+            rel="noopener noreferrer"
+          >
+            Directions(Google Maps)
```
[the remark this answers is not in the document]
Comment (chriskim2311) [reply src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx:33]: Solved this with a different text confirmed with Jordan

File: src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx (hunk 2 above)
Comment (chriskim2311) [reply src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx:20]: Good catch

File: src/applications/static-pages/events/helpers/index.unit.spec.js (hunk 4 above)
Comment (chriskim2311) [reply src/applications/static-pages/events/helpers/index.unit.spec.js:318]: This helper no longer renders this field, its an optional field in the CMS

**Review by randimays — APPROVED** [review]
LGTM!

**Comment by laflannery** [comment]
@chriskim2311 a couple things:
- The directions links should say "Get directions on Google Maps" as we have in the Sketch files:
image
- Also, I can't currently get into the review instance (it might still be deploying?) but can you confirm if there is `sr-only` text on those links? Like we have on VAMC Locations links:
image

**Comment by thejordanwood** [comment]
@chriskim2311 I agree with Laura that the directions links should say "Get directions on Google Maps". Everything else looks good!

**Pull request merged**